netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -105,6 +128,59 @@ public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData
     });
   }
 
+  /**
+   * Sends a ping to the server.
+   */
+  void ping(final PingCallback callback, Executor executor) {
```

Comment:
Rather than exposing this method, could we just have the transport send the `SendPingCommand` and then have all of this done in the transport thread? That would be more consistent with the current interface between the transport/stream and the handler.